### PR TITLE
Celery 4 (attempt 2)

### PR DESCRIPTION
**Pull request by leohemsted**
Brought in martyn's changes - had to do some hefty conflict resolution. Got rid of his definitions.py change cos it was just a bit too big.

* There are some *gross* problems related to imports. I'd like to sort them out before 
* Priority queue locally seems absolutely fine. sent_at - created_at is ~99% < 1 second. Am seeing prod-like times when running celery 3 (15, 20 seconds etc).

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -26,6 +25,11 @@
 DATE_FORMAT = "%Y-%m-%d"
 
 db = SQLAlchemy()
+
+# avoid circular import by importing after db
+from app.celery.celery import NotifyCelery
+
```
Comment (imdadahad) [inline app/__init__.py]: You can move this back above (as discussed).

File: app/celery/celery.py [hunk 2]
```diff
@@ -1,11 +1,130 @@
...
+
+class CeleryConfig:
+    def __init__(self, config):
+        self.broker_transport_options['queue_name_prefix'] = config['NOTIFICATION_QUEUE_PREFIX']
+        self.broker_url = config.get('BROKER_URL', 'sqs://')
+
+    broker_transport_options = {
+        'region': 'sqs.eu-west-1',
+        'polling_interval': 1,  # 1 second
+        'visibility_timeout': 310,
```
Comment (imdadahad) [inline app/celery/celery.py]: Do we still need this `polling_interval' if now we're now using long polling.

Comment (leohemsted) [inline app/celery/celery.py]: ¯\_(ツ)_/¯

Comment (leohemsted) [inline app/celery/celery.py]: long polling means that SQS will keep the request open until it gets a message - but after it returns, celery still needs to know how long to wait until asking for the next one - this is the `polling_interval` (i think)

Comment (imdadahad) [inline app/celery/celery.py]: 👍

**Comment by imdadahad** [comment]
This needs `pycurl` as I get `The curl client requires the pycurl library` when trying to start celery.

**Pull request merged**